.github/actions/execute-runner/index.js

```diff
@@ -0,0 +1,173 @@
+// Copyright 2014-2021 Scalyr Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const core = require('@actions/core');
+const cache = require('@actions/cache');
+const fs = require('fs');
+const path = require('path')
+const child_process = require('child_process')
+const buffer = require('buffer')
+const readline = require('readline')
+
+
+async function checkAndGetCache(
+    name,
+    cacheDir,
+    finalCacheSuffix
+) {
+    //
+    // Check if there is an existing cache for the given step name.
+    //
+    const cachePath = path.join(cacheDir, name)
+    const key = `${name}-${finalCacheSuffix}`
+
+    // try to restore the cache.
+    const result = await cache.restoreCache([cachePath], key)
+
+    if (typeof result !== "undefined") {
+        console.log(`Cache for the step with key ${key} is found.`)
+    } else {
+        console.log(`Cache for the step with key ${key} is not found.`)
+    }
+
+    // Return whether it is a hit or not.
+    return result
+}
+
+async function checkAndSaveCache(
+    name,
+    cacheDir,
+    isHit,
+    finalCacheSuffix
+) {
+    //
+    // Save the cache directory for a particular step if it hasn't been saved yet.
+    //
+
+    const fullPath = path.join(cacheDir, name)
+    const cacheKey = `${name}-${finalCacheSuffix}`
+
+    // Skip files. Cache can be only the directory.
+    if (!fs.lstatSync(fullPath).isDirectory()) {
+        return
+    }
+
+    // If there's no cache hit, then save directory to the cache now.
+    if (isHit) {
+        console.log(`Cache for the step with key ${cacheKey} has been hit. Skip saving.`)
```

Comment:
Perhaps may make it a bit easier to understand if it said something along the lines of "Cache for key already exist, skip saving".

Since for a second I was bit confused what hit means in this context (I assumed it means cache for that key has been found / exists, but I was still confused for a second).